# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1458166379 25200
#      Wed Mar 16 15:12:59 2016 -0700
# Node ID c1c7748b00fe7661b4060bb234ae4f664aad3d8f
# Parent  9d1a5b22e87ad8112af1602a162d2184ee4c53ba
system.tst: Fix failing BIST test if /dev/core does not exist on UNIX system

* system.tst: Use exist() to check whether the /dev/core device file exists
before running S_ISLNK test.

diff --git a/test/system.tst b/test/system.tst
--- a/test/system.tst
+++ b/test/system.tst
@@ -182,17 +182,19 @@
 %!error <Invalid call to lstat> lstat ("foo", 1)
 
 %!test
 %! if (isunix ())
 %!   assert (S_ISCHR (stat ("/dev/null").mode));
 %!   if (exist ("/dev/initctl"))
 %!     assert (S_ISFIFO (stat ("/dev/initctl").mode));
 %!   endif
-%!   assert (S_ISLNK (lstat ("/dev/core").mode));
+%!   if (exist ("/dev/core"))
+%!     assert (S_ISLNK (lstat ("/dev/core").mode));
+%!   endif
 %! endif
 %! nm = tempname ();
 %! fid = fopen (nm, "wb");
 %! fclose (fid);
 %! r = [ S_ISREG(stat(nm).mode)
 %!       S_ISDIR(stat(nm).mode)
 %!       S_ISCHR(stat(nm).mode)
 %!       S_ISBLK(stat(nm).mode)
